modules/Microsoft.App/managedEnvironments/deploy.bicep

```diff
@@ -0,0 +1,171 @@
+param infrastructureSubnetId string = ''
+
+@description('Optional. Boolean indicating the environment only has an internal load balancer. These environments do not have a public static IP resource. They must provide runtimeSubnetId and infrastructureSubnetId if enabling this property.')
+param internal bool = false
+
+@description('Optional. Configuration used to control the Environment Egress outbound traffic.')
+param vnetOutboundSettings object = {}
+
+@description('Optional. IP range in CIDR notation that can be reserved for environment infrastructure IP addresses. Must not overlap with any other provided IP ranges.')
+param platformReservedCidr string = ''
+
+@description('Optional. An IP address from the IP range defined by platformReservedCidr that will be reserved for the internal DNS server.')
+param platformReservedDnsIP string = ''
+
+@description('Optional. Resource ID of a subnet that Container App containers are injected into. This subnet must be in the same VNET as the subnet defined in infrastructureSubnetId. Must not overlap with any other provided IP ranges.')
+param runtimeSubnetId string = ''
+
+@description('Optional. Whether or not this Managed Environment is zone-redundant.')
+param zoneRedundant bool = false
+
+@description('Optional. Password of the certificate used by the custom domain.')
+@secure()
+param certificatePassword string = ''
+
+@description('Optional. Certificate to use for the custom domain. PFX or PEM.')
+@secure()
+param certificateValue string = ''
+
+@description('Optional. DNS suffix for the environment domain.')
+param dnsSuffix string = ''
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@description('Optional. Workload profiles configured for the Managed Environment.')
+param workloadProfiles array = []
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2022-09-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name, location)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2022-10-01' existing = {
+  name: last(split(logAnalyticsWorkspaceResourceId, '/'))!
+  scope: resourceGroup(split(logAnalyticsWorkspaceResourceId, '/')[2], split(logAnalyticsWorkspaceResourceId, '/')[4])
+}
+
+resource managedEnvironment 'Microsoft.App/managedEnvironments@2022-06-01-preview' = {
+  name: name
+  location: location
+  tags: tags
+  sku: {
+    name: skuName
+  }
+  properties: {
+    appLogsConfiguration: {
+      destination: logsDestination
+      logAnalyticsConfiguration: {
```

Comment:
This probably needs a condition if the logAnalyticsWorkspaceResourceID is ''

Reply:
Added

Reply:
As discussed, we'll "unfortunately" have to mandate the Log Analytics as a requirement for the module and open up an issue for the limitation on the RP. I will link the issue once I draft it

Reply:
Error code

```powershell
(Code: InvalidTemplateDeployment)
 - Validation failed for a resource. Check 'Error.Details[0]' for more information. (Code: ValidationForResourceFailed)
   - Invalid request body for environment. Path: $.appLogsConfiguration. Does not conform to Managed Environment schema, please visit for more information https://docs.microsoft.com/azure/container-apps/azure-resource-manager-api-spec?tabs=arm-template#container-apps-environment (Code:ManagedEnvironmentInvalidSchema)
  ```

Reply:
Added a reference to bug  https://github.com/Azure/bicep/issues/9990